numba/dppl/spirv_generator.py

```diff
@@ -116,6 +124,17 @@ def finalize(self):
         spirv_path = self._track_temp_file("generated-spirv")
         self._cmd.generate(ipath=self._llvmfile, opath=spirv_path)
 
+        binary_paths = [spirv_path]
+        for key in list(self.context.link_binaries.keys()):
+            del self.context.link_binaries[key]
+            if key == LINK_ATOMIC:
```

Comment:
I wanted to have the option of being able to use it later, as far as I know set is implemented using a hashmap so we will not be saving much here (https://stackoverflow.com/questions/3949310/how-is-set-implemented). Please let me know what you think.

Reply:
Performance is not what I'm worriyng about.
It's more about case, when someone would wrote code like:
```python
link_libraries[LINK_ATOMIC] = False
```

But it would still link with the library

Reply:
I agree in general but in this case the future use case is not True or False (at least not now). Currently the key being there indicates we want to link which is why we are also removing the key as soon as we are done.